R/dataPrepRast.R

```diff
@@ -0,0 +1,131 @@
+#' Crop, reproject, and mask a raster map
+#'
+#' This script reprojects and crop a raster using a raster or polygon object.
+#'
+#' @param x   A \code{RasterLayer} object or the name of one.
+#' @param cropWith   Extent object, or any object from which an Extent object can
+#'                   be extracted (see Details of \code{\link[raster]{crop}}).
+#' @params maskWith   Raster* or Spatial object passed to \code{\link[raster]{mask}}.
+#' @params reprojectTo   Raster* or Spatial object passed to \code{\link[raster]{crs}}.
+#' @params outPath   Path where fileOut is saved.
+#' @params fileOut   File name to write to (\code{\link[raster]{writeRaster}}).
+#' @params formatOut File format to write out (\code{\link[raster]{writeRaster}}).
+#'                   If not specified, GTiff will be used.
+#' @params timings   Logical indicating whether Sys.time messages should be
+#'                   suppressed. Default is \code{FALSE}.
+#' @param ...   Additional arguments to the aforementioned \code{raster} functions.
+#'
+#' @return Invisibly returns the cropped, projected, and masked raster.
+#'
+#' @importFrom raster compareCRS crop mask crs projectRaster writeRaster
+#' @importFrom sp spTransform
+#' @export
+#' @docType methods
+#' @examples
+#' library(raster)
+#' library(maptools)
+#' data("wrld_simpl")
+#'
+#' # Set
+#' outputPath <- tempdir()
+#'
+#' #From scratch, random raster
+#' xy <- matrix(rnorm(2500),50,50)
+#' inputrast <- raster(xy)
+#'
+#' #Set Canada extent and projection
+#' extent(inputrast) <- c(-170,-15,25,100)
+#' projection(inputrast) <- CRS("+proj=longlat +datum=WGS84")
+#'
+#' #Extract Canada boundary from wrld_simpl
+#' canada<-wrld_simpl[which(wrld_simpl@data$NAME =='Canada'),]
+#' #dataPrepRast
+#' x<-dataPrepRast(x=inputrast, cropWith=canada , maskWith= canada,
+#'                  reprojectTo= inputrast, timings = TRUE)
+dataPrepRast <- function(x, cropWith=NULL, maskWith=NULL, reprojectTo=NULL, outPath=NULL,
+                   fileOut=NULL, formatOut=NULL, timings=FALSE, ...) {
+  wd<-getwd()
```

Comment:
why do you set default `formatOut` to `NULL` here, when GTiff is default (which you set later)?